bin/spice/cmd/upgrade.go

```diff
@@ -68,14 +68,21 @@ spice upgrade
 			cleanupOldBinaries()
 		}
 
+		slog.Info("Checking for latest Spice Runtime release...")
+		currentVersion, err := rtcontext.Version()
+		if err != nil {
+			slog.Info("Spice runtime is not installed and won't be upgraded. Run `spice install` to install the runtime.")
+			return
+		}
+
 		runtimeUpgradeRequired, err := rtcontext.IsRuntimeUpgradeAvailable()
 		if err != nil {
 			slog.Error("checking for runtime upgrade", "error", err)
```

Comment:
It seems we have an issue to update this logic
https://github.com/spiceai/spiceai/issues/4657